web/pandas/pdeps/0009-io-extensions.md

```diff
@@ -0,0 +1,404 @@
+# PDEP-9: Allow third-party projects to register pandas connectors with a standard API
+
+- Created: 5 March 2023
+- Status: Under discussion
+- Discussion: [#51799](https://github.com/pandas-dev/pandas/pull/51799)
+              [#53005](https://github.com/pandas-dev/pandas/pull/53005)
+- Author: [Marc Garcia](https://github.com/datapythonista)
+- Revision: 1
+
+## PDEP Summary
+
+This document proposes that third-party projects implementing I/O or memory
+connectors to pandas can register them using Python's entrypoint system,
+and make them available to pandas users with the usual pandas I/O interface.
+For example, packages independent from pandas could implement readers from
+DuckDB and writers to Delta Lake, and when installed in the user environment
+the user would be able to use them as if they were implemented in pandas.
+For example:
+
+```python
+import pandas
+
+pandas.load_io_plugins()
```

Comment:
Can we make this explicit (require the user specify which io plugin to load)?

Reply:
We can, but I don't see how this is much different than using `import pandas_myformat`, which we can already do without the need of any change in pandas. We could still try to enforce a standard API, but we lose the easiness of use in my opinion.